torchinductor/codegen/triton.py

```diff
@@ -735,7 +735,12 @@ def load(self, name: str, index: sympy.Expr):
             ep = ", eviction_policy='evict_last'"
         else:
             ep = ""
-        line = f"tl.load({var} + {index}, {mask}{ep})"
+
+        if V.graph.is_unspec_arg(name):
+            line = var
```

Comment:
@ngimel, you mentioned this may be a problem for ternary ops? or is it the other way around, 0-dim tensor is a problem for ternary ops?

Reply:
That's a problem for ternary ops in eager, but if we codegen them that should be fine. For all ops, the values will be baked in as constants though, so we should be guarding on them.